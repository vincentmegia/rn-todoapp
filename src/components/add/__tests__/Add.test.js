import React from 'react'
import Add from '..'
import {render} from '@testing-library/react-native'
import {act} from 'react-test-renderer'

describe('Add component tests', () => {
  it('it matches snapshot', () => {
    const {toJSON} = render(
      <Add
        visible={true}
        onButtonPress={jest.fn()}
        onBackdropPress={jest.fn()}
      />,
    )
    expect(toJSON()).toMatchSnapshot()
  })

  it('it publishes new todos to parent', () => {
    const mockedButtonPress = jest.fn()
    const mockedBackdropPress = jest.fn()
    const {getByTestId} = render(
      <Add
        visible={true}
        onButtonPress={mockedButtonPress}
        onBackdropPress={mockedBackdropPress}
      />,
    )
    act(() => {
      getByTestId('add_input').props.onChangeText('first to do')
      getByTestId('new_button').props.onClick()
    })

    expect(mockedButtonPress).toHaveBeenCalledTimes(1)
  })
})
