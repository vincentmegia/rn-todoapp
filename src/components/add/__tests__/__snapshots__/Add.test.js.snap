// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add component tests it matches snapshot 1`] = `
<Modal
  hardwareAccelerated={false}
  onRequestClose={[MockFunction]}
  testID="modal"
  transparent={true}
  visible={true}
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(0, 0, 0, .4)",
        "height": "100%",
        "justifyContent": "center",
        "left": 0,
        "position": "absolute",
        "top": 0,
        "width": "100%",
      }
    }
    testID="RNE__Overlay__backdrop"
  />
  <View
    pointerEvents="box-none"
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
    testID="RNE__Overlay__Container"
  >
    <View
      style={
        Object {
          "backgroundColor": "rgb(255, 255, 255)",
          "borderRadius": 10,
          "padding": 10,
          "shadowColor": "rgba(0, 0, 0, .3)",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowRadius": 4,
        }
      }
      testID="RNE__Overlay"
    >
      <View
        style={
          Object {
            "marginVertical": 16,
            "width": 700,
          }
        }
      >
        <View
          style={
            Object {
              "paddingHorizontal": 10,
              "width": "100%",
            }
          }
          testID="RNE__Input__view-wrapper"
        >
          <Text
            style={
              Object {
                "color": "#86939e",
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            Add todo
          </Text>
          <View
            collapsable={false}
            style={
              Object {
                "alignItems": "center",
                "borderBottomWidth": 1,
                "borderColor": "#86939e",
                "flexDirection": "row",
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                ],
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "height": 40,
                  "justifyContent": "center",
                  "marginVertical": 4,
                  "paddingRight": 4,
                }
              }
            >
              <View
                style={
                  Object {
                    "marginRight": 8,
                    "overflow": "hidden",
                  }
                }
                testID="RNE__ICON__CONTAINER"
              >
                <View
                  accessibilityRole="button"
                  android_ripple={
                    Object {
                      "borderless": false,
                      "color": "rgba(0, 0, 0, 0.3)",
                      "radius": -5,
                    }
                  }
                  disabled={false}
                  testID="RNE__ICON__CONTAINER_ACTION"
                >
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "justifyContent": "center",
                      }
                    }
                    testID="RNE__ICON"
                  >
                    <Text
                      allowFontScaling={false}
                      selectable={false}
                      style={
                        Array [
                          Object {
                            "color": "grey",
                            "fontSize": 20,
                          },
                          Object {
                            "backgroundColor": "transparent",
                          },
                          Object {
                            "fontFamily": "Material Icons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                      testID="RNE__ICON__Component"
                    >
                      
                    </Text>
                  </View>
                </View>
              </View>
            </View>
            <TextInput
              editable={true}
              onChangeText={[Function]}
              placeholderTextColor="#86939e"
              style={
                Object {
                  "color": "#242424",
                  "flex": 1,
                  "fontSize": 18,
                  "minHeight": 40,
                }
              }
              testID="add_input"
              underlineColorAndroid="transparent"
            />
          </View>
          <Text
            style={
              Object {
                "color": "#ff190c",
                "fontSize": 12,
                "margin": 5,
              }
            }
          />
        </View>
        <View
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 8,
              },
              Object {
                "alignSelf": "center",
                "width": 150,
              },
              false,
            ]
          }
          testID="RNE_BUTTON_WRAPPER"
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              Object {
                "busy": false,
                "disabled": false,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID="new_button"
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#2089dc",
                  "borderColor": "#2089dc",
                  "borderRadius": 8,
                  "borderWidth": 0,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "padding": 8,
                  "paddingHorizontal": 10,
                  "width": 150,
                }
              }
            >
              <View
                style={
                  Object {
                    "overflow": "hidden",
                  }
                }
                testID="RNE__ICON__CONTAINER"
              >
                <View
                  accessibilityRole="button"
                  android_ripple={
                    Object {
                      "borderless": false,
                      "color": "rgba(0, 0, 0, 0.3)",
                      "radius": -5,
                    }
                  }
                  disabled={false}
                  testID="RNE__ICON__CONTAINER_ACTION"
                >
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "justifyContent": "center",
                      }
                    }
                    testID="RNE__ICON"
                  >
                    <Text
                      allowFontScaling={false}
                      selectable={false}
                      style={
                        Array [
                          Object {
                            "color": "white",
                            "fontSize": 25,
                          },
                          Object {
                            "backgroundColor": "transparent",
                            "marginRight": 10,
                          },
                          Object {
                            "fontFamily": "FontAwesome",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                      testID="RNE__ICON__Component"
                    >
                      
                    </Text>
                  </View>
                </View>
              </View>
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 18,
                    "paddingVertical": 1,
                    "textAlign": "center",
                  }
                }
              >
                Save
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
